src/main/kotlin/device/application/GetDevice.kt

```diff
@@ -0,0 +1,16 @@
+package device.application
+
+import device.domain.Device
+import device.domain.DeviceIdentifier
+import device.domain.DeviceRepository
+
+class GetDevice(
+    private val repository: DeviceRepository,
+) {
+    /***
+     * Searches a Device with the given identifier and returns it if found.
+     * @see Device
+     * @return null or a device instance if it exists
+     */
+    fun get(identifier: DeviceIdentifier) = repository.get(identifier)
```

Comment:
This should receive a primitive or a DevicePrimitives object:
```kotlin
    fun get(primitives: DevicePrimitives) = repository.get(Identifier(primitives.identifier))
```
or
```kotlin
    fun get(identifier: String) = repository.get(Identifier(identifier))
```

Reply:
Get function shouldn't receive a primitives object, but a Identifier, as that's the only thing we really want.
That would be a good practice to apply on the create and update use case.
Is that correct?

Reply:
I've refactored the create use case as we said.